conans/test/integration/command/remote_test.py

```diff
@@ -346,50 +345,56 @@ def test_add_wrong_conancenter():
-        self.client.run('remote update default --filter="app/*" --filter="liba/2.*"')
-
-        self.client.run("install --requires=app/1.0 -r=default", assert_error=True)
-        assert "app/1.0: Downloaded recipe revision 04ab3bc4b945a2ee44285962c277906d" in self.client.out
-        assert "liba/1.0: Downloaded recipe revision 4d670581ccb765839f2239cc8dff8fbd" not in self.client.out
-        assert "ERROR: Package 'liba/1.0' not resolved: Unable to find 'liba/1.0' in remotes" in self.client.out
-        self.client.run("remove * -c")
-
-        self.client.run('remote update default --filter="liba/*" --filter="app/*"')
-        self.client.run("install --requires=app/1.0 -r=default", assert_error=True)
-        assert "app/1.0: Downloaded recipe revision 04ab3bc4b945a2ee44285962c277906d" in self.client.out
-        assert "liba/1.0: Downloaded recipe revision 4d670581ccb765839f2239cc8dff8fbd" in self.client.out
-        assert "ERROR: Package 'libb/[>=1.0]' not resolved" in self.client.out
-        self.client.run("remove * -c")
-
-        self.client.run('remote update default --filter="liba/*" --filter="app/*"')
-        self.client.run("install --requires=app/1.0 -r=default", assert_error=True)
-        assert "app/1.0: Downloaded recipe revision 04ab3bc4b945a2ee44285962c277906d" in self.client.out
-        assert "liba/1.0: Downloaded recipe revision 4d670581ccb765839f2239cc8dff8fbd" in self.client.out
-        assert "ERROR: Package 'libb/[>=1.0]' not resolved" in self.client.out
-        self.client.run("remove * -c")
-
-        self.client.run('remote update default --filter="*"')
-        self.client.run("install --requires=app/1.0 -r=default")
-        assert "app/1.0: Downloaded recipe revision 04ab3bc4b945a2ee44285962c277906d" in self.client.out
-        assert "liba/1.0: Downloaded recipe revision 4d670581ccb765839f2239cc8dff8fbd" in self.client.out
-        assert "libb/1.0: Downloaded recipe revision 4d670581ccb765839f2239cc8dff8fbd" in self.client.out
+def test_allowed_packages_remotes():
+    tc = TestClient(light=True, default_server_user=True)
+    tc.save({"conanfile.py": GenConanfile(),
+             "app/conanfile.py": GenConanfile("app", "1.0")
+            .with_requires("liba/1.0")
+            .with_requires("libb/[>=1.0]")})
+    tc.run("create . --name=liba --version=1.0")
+    tc.run("create . --name=libb --version=1.0")
+    tc.run("create app")
+    tc.run("upload * -r=default -c")
+    tc.run("remove * -c")
+
+    tc.run("install --requires=app/1.0 -r=default")
+    assert "app/1.0: Downloaded recipe revision 04ab3bc4b945a2ee44285962c277906d" in tc.out
+    assert "liba/1.0: Downloaded recipe revision 4d670581ccb765839f2239cc8dff8fbd" in tc.out
+    tc.run("remove * -c")
+
+    # lib/2.* pattern does not match the one being required by app, should not be found
+    tc.run('remote update default --allowed-packages="app/*" --allowed-packages="liba/2.*"')
```

Comment:
If I have a relatively long list, maybe it makes sense to also have a short `-a=` too?